easybuild/easyblocks/s/score_p.py

```diff
@@ -38,49 +39,83 @@
+                toolchain.GCC: 'gcc',
+                toolchain.IBMCOMP: 'ibm',
+                toolchain.INTELCOMP: 'intel',
+                toolchain.PGI: 'pgi',
+            }
+            comp_fam = self.toolchain.comp_family()
+            if comp_fam in comp_opts:
+                self.cfg.update('configopts', "--with-nocross-compiler-suite=%s" % comp_opts[comp_fam])
             else:
-                raise EasyBuildError("MPI family %s not supported yet (only: %s)",
-                                     mpi_fam, ', '.join(mpi_opts.keys()))
+                raise EasyBuildError("Compiler family %s not supported yet (only: %s)",
+                                     comp_fam, ', '.join(comp_opts.keys()))
+
+            # --with-mpi=(bullxmpi|hp|ibmpoe|intel|intel2|intelpoe|lam|mpibull2|mpich|mpich2|mpich3|openmpi|
+            #             platform|scali|sgimpt|sun)
+            #
+            # Notes:
+            #   - intel:    Intel MPI v1.x (ancient & unsupported)
+            #   - intel2:   Intel MPI v2.x and higher
+            #   - intelpoe: IBM POE MPI for Intel platforms
+            #   - mpich:    MPICH v1.x (ancient & unsupported)
+            #   - mpich2:   MPICH2 v1.x
+            #   - mpich3:   MPICH v3.x & MVAPICH2
+            #               This setting actually only affects options passed to the MPI (Fortran) compiler wrappers.
+            #               And since MPICH v3.x-compatible options were already supported in MVAPICH2 v1.7, it is
+            #               safe to use 'mpich3' for all supported versions although MVAPICH2 is based on MPICH v3.x
+            #               only since v1.9b.
+            #
+            # With minimal toolchains, packages using this easyblock may be built with a non-MPI toolchain (e.g., OTF2).
+            # In this case, skip passing the '--with-mpi' option.
+            mpi_opts = {
+                toolchain.INTELMPI: 'intel2',
+                toolchain.OPENMPI: 'openmpi',
+                toolchain.MPICH: 'mpich3',     # In EB terms, MPICH means MPICH 3.x
+                toolchain.MPICH2: 'mpich2',
+                toolchain.MVAPICH2: 'mpich3',
+            }
+            mpi_fam = self.toolchain.mpi_family()
+            if mpi_fam is not None:
+                if mpi_fam in mpi_opts:
+                    self.cfg.update('configopts', "--with-mpi=%s" % mpi_opts[mpi_fam])
+                else:
+                    raise EasyBuildError("MPI family %s not supported yet (only: %s)",
+                                         mpi_fam, ', '.join(mpi_opts.keys()))
 
-        # auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
+        # Auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
         deps = {
-            'binutils': ['--with-libbfd=%%s/%s' % get_software_libdir('binutils', fs=['libbfd.a'])],
+            'binutils': ['--with-libbfd=%s'],
```

Comment:
@geimer The original PR for the Score-P easyblock (https://github.com/hpcugent/easybuild-easyblocks/pull/304) mentions the need for specifying the `lib` subdir for `--with-libbfd` (and `--with-papi-lib`) rather than specifying the root, so it seems like we did have a good reason for this...

Could this have been a bug in the handling of `--with-libbfd` option in Score 1.2.x, and this behaviour being retained as a fallback in later versions for backward compatibility?

Maybe @berndmohr remembers why we needed to specify the path to the `/lib` subdirectory of the `binutils` install directory rather than the root install prefix as suggested with the help output for `--with-libbfd`?

My suggestion would be to put a version check in place for versions older than Score-P 1.4, and stick to using the `/lib` approach for those, to avoid breaking stuff?